Rework Prism class & other minor changes

LXGaming opened the pull request:
**Minor Changes**:
- Updated Gradle to v4.10.3
- Removed Messages class as it was made redundant by #112 
- Removed the game variable from Prism class in favor of Sponge methods
- Add Reference class for storing constant variables
- Replaced `http://discover-prism.com` with `https://github.com/prism/Prism/wiki` as it's no longer valid nor can be trusted as anyone could claim it

**Major Changes**:
- Rewrite Prism class
Prism now loads throughout Sponges lifecycle instead of loading everything at once on StartedServer
This should improve 3rd party plugin support that use Prisms API
Also improves code readability and maintainability (for me personally at least)

- Rewrite build.gradle
Update License and Shadow Gradle plugins
Easier to add Forge Mod support later if required
Primarily to improve readability and maintainability for me personally

viveleroi commented:
Looks good to me. My only request is can you trim trailing whitespace? I noticed a few empty lines with tab additions in the diff.

After that, are you ready for merge?

viveleroi commented:
Also, I've invited you to be a collaborator on this repo, which will allow you post releases, etc.

My only request is that you still PR changes so I can at least review them.

The sponge forum login is down right, hence why I'm saying this here.

LXGaming commented:
I've removed the whitespace on empty lines from all class files (I've updated my IDE settings to prevent it from reoccurring)

Yes this is ready to merge.
